### ISSUE TITLE
Korean words being seperated after ios 17 updates 

### ISSUE THREAD

**Issue by YoungrakSon** [flutter/flutter#135825]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Write Korean words in Flutter TextFields

### Expected results

검색이 안됨

It is supposed to be like this.

### Actual results

ㄱㅓㅁㅅㅐㄱㅇㅣ ㅇㅏㄴㄷㅗㅣ

If you type korean words to flutter textfield words doesn't combine and you cannot use flutter app.. 

It happens after ios 17 updates. 

### Code sample

![Simulator Screenshot - iPhone 14 Pro - 2023-10-01 at 22 00 50](https://github.com/flutter/flutter/assets/37031340/3d7aa72f-a323-4ba3-9bf0-1d7db088acc9)

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by YoungrakSon** [comment]
Please help..

**Comment by huycozy** [comment]
Reproduced this issue on iPhone 14 Pro, iOS 17.0 whilst it works as expected on iPhone 7, iOS 15.7.9. Thanks for the report!

<details>
<summary>Demo</summary>

| iOS 17.0 | iOS 15.7.9 |
| --------------- | --------------- |
<video src="https://github.com/flutter/flutter/assets/104349824/3f43731c-b967-44cb-a38b-3da18feaaa2e"/> | <video src="https://github.com/flutter/flutter/assets/104349824/d358124a-50a7-4753-8dfb-7f8fa5567f77"/>

</details>

<details>
<summary>flutter doctor -v (stable and master)</summary>

```bash
[✓] Flutter (Channel stable, 3.13.6, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.13.6 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ead455963c (5 days ago), 2023-09-26 18:28:17 -0700
    • Engine revision a794cf2681
    • Dart version 3.1.3
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.82.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 117.0.5938.132

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

```bash
[!] Flutter (Channel master, 3.15.0-13.0.pre.40, on macOS 14.0 23A344 darwin-x64, locale en-VN)
    • Flutter version 3.15.0-13.0.pre.40 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    ! Warning: `flutter` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/flutter, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    ! Warning: `dart` on your path resolves to /Users/huynq/Documents/GitHub/flutter/bin/dart, which is not inside your current Flutter SDK checkout at /Users/huynq/Documents/GitHub/flutter_master. Consider adding /Users/huynq/Documents/GitHub/flutter_master/bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 71acf24201 (3 hours ago), 2023-10-02 01:45:02 +0200
    • Engine revision 10b930ad1a
    • Dart version 3.2.0 (build 3.2.0-220.0.dev)
    • DevTools version 2.28.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly to perform update checks and upgrades.

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-34, build-tools 32.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio Giraffe.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 15.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 15A240d
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio Giraffe.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.82.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.72.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 14.0 23A344 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 117.0.5938.132

[✓] Network resources
    • All expected network resources are available.

! Doctor found issues in 1 category.
```
</details>

**Comment by LongCatIsLooong** [comment]
@YoungrakSon are you seeing this on a real device or simulator? This appears to be an iOS keyboard bug since the jamos aren't composing in UIKit apps either.

**Comment by urim-snu** [comment]
Currently, users of our service are experiencing the same thing. We developed the app with flutter version 3.13.1, and the users are all real device users and their OS is either ios 17.0 or 17.0.3.

@LongCatIsLooong

**Comment by LongCatIsLooong** [comment]
> Currently, users of our service are experiencing the same thing. We developed the app with flutter version 3.13.1, and the users are all real device users and their OS is either ios 17.0 or 17.0.3.

Could you confirm that the Korean virtual keyboard is working in UIKit apps (for example apps that came with the os like reminder)? It has the same problem for me when I tried using a simulator running iOS 17.0.

**Comment by huycozy** [comment]
@LongCatIsLooong I checked this on iOS native app (Reminder) and a pure native TextField, the issue also appears there on iOS 17 simulator:

<details>
<summary>Demo</summary>

https://github.com/flutter/flutter/assets/104349824/21c81785-e822-4fb1-a7f6-addb4fba05c9
</details>

**Comment by YoungrakSon** [comment]
> @YoungrakSon are you seeing this on a real device or simulator? This appears to be an iOS keyboard bug since the jamos aren't composing in UIKit apps either.

It was both on simulator and real device.

my real device is Iphone 14 pro and using simulator Iphone 14 pro as well.  
I tried Iphone 15 simulator and same problem. 


Thank you for your generous help.

**Comment by leehack** [comment]
I see the same issue on Safari.app
Not sure if the issue has been reported to Apple but It doesn't seem Flutter issue.
<img width="466" alt="image" src="https://github.com/flutter/flutter/assets/1233094/4e93a1a3-a6e7-487f-bbda-4561337bfbe7">

**Comment by LongCatIsLooong** [comment]
This is fixed on iOS 17.1. Closing.
